components/contact-list.tsx

```diff
@@ -21,13 +22,18 @@ import Link from "next/link";
 type ContactListProps = {
   data: Contact[];
   provinceSlug: string;
+  loading: boolean;
```

Comment:
```suggestion
  isLoading: boolean;
```

IMO, better to use the `is-` prefix, to better follow the [A/HC/LC naming pattern](https://github.com/kettanaito/naming-cheatsheet#ahclc-pattern).

Same with the `useSearch()` hook, instead of returning `loading`, make it `isLoading` too.